### ISSUE TITLE
ClassNotFoundException: org.eclipse.emf.mwe2.runtime.workflow.IWorkflowComponent

### ISSUE THREAD

**Issue by kthoms** [xtext/xtext-gradle-plugin#100]
I get a strange CNFE for IWorkflowComponent by just changing the Xtext version on a project to >2.10. With 2.10 the build succeeds.

```
g build --stacktrace

:org.xtext.example.mydsl.model:generateXtext FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':org.xtext.example.mydsl.model:generateXtext'.
> org/eclipse/emf/mwe2/runtime/workflow/IWorkflowComponent

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':org.xtext.example.mydsl.model:generateXtext'.
   [...]
Caused by: java.lang.NoClassDefFoundError: org/eclipse/emf/mwe2/runtime/workflow/IWorkflowComponent
        at org.xtext.gradle.builder.XtextGradleBuilder.<init>(XtextGradleBuilder.java:97)
        at org.xtext.gradle.builder.XtextGradleBuilderFactory.get(XtextGradleBuilderFactory.java:14)
        at org.xtext.gradle.tasks.internal.IncrementalXtextBuilderProvider.createBuilder(IncrementalXtextBuilderProvider.java:80)
        at org.xtext.gradle.tasks.internal.IncrementalXtextBuilderProvider.getBuilder(IncrementalXtextBuilderProvider.java:39)
        at org.xtext.gradle.tasks.XtextGenerate.initializeBuilder(XtextGenerate.java:417)
        at org.xtext.gradle.tasks.XtextGenerate.generate(XtextGenerate.java:146)
```

Full log:
[log.txt](https://github.com/xtext/xtext-gradle-plugin/files/1518010/log.txt)

The problem can be reproduced with this project: https://github.com/itemis/itemis-blog/tree/xcore_gradle_upgrade_xtext

In the main build script, if `ext.xtextVersion` is set to 2.10.0, the build works. The plugin breaks when 2.11.0 or higher is used.

I have verified that org.eclipse.emf.mwe2.runtime is on the plugin classpath (use the `printBuildScriptClasspath` task from the script).

The project uses Gradle 3.1, but same problem for 4.3.1.

**Comment by cdietrich** [comment]
which version of the plugin do you use?

**Comment by kthoms** [comment]
This is the full debug log
[debug-log.txt](https://github.com/xtext/xtext-gradle-plugin/files/1518184/log.txt)

Suspicious: In line 2650 the generateXtextTask is started:
```
Starting to execute task ':org.xtext.example.mydsl.model:generateXtext'
```

There is no `org.eclipse.emf.mwe2.runtime` mentioned anywhere.

**Comment by cdietrich** [comment]
@oehme is the plugin configs somehow share between diffeent modules?

**Comment by cdietrich** [comment]
generator.outlet.producesJava = true leads to the issue. still have no clue why

**Comment by cdietrich** [comment]
its actually missing trnasitive dependencies that were provided by xtext previously.

**Comment by oehme** [comment]
Seems like xtext's standalone builder leaks a dependency on `IWorkFlowComponent`. It's not used anywhere directly in this plugin (I checked the generated Java code).

**Comment by oehme** [comment]
Thanks, closing as not related to the plugin.

**Comment by AresEkb** [comment]
I have the same exception. Could you please point me out how to fix it? I have no idea how to add this dependency to a build script. It doesn't work:

```
dependencies {
    compile "org.eclipse.xtext:org.eclipse.xtext:${xtextVersion}"
    compile "org.eclipse.xtext:org.eclipse.xtext.xbase:${xtextVersion}"
    compile 'org.eclipse.emf:org.eclipse.emf.ecore.xcore.lib:+'

    xtextLanguages 'org.eclipse.emf:org.eclipse.emf.ecore.xcore:+'
    xtextLanguages 'org.eclipse.emf:org.eclipse.emf.ecore.xcore.lib:+'
    xtextLanguages 'org.eclipse.emf:org.eclipse.emf.codegen:+'
    xtextLanguages 'org.eclipse.emf:org.eclipse.emf.codegen.ecore:+'
    xtextLanguages 'org.eclipse.emf:org.eclipse.emf.codegen.ecore.xtext:+'
    xtextLanguages "org.eclipse.xtext:org.eclipse.xtext.ecore:${xtextVersion}"
    xtextLanguages 'org.eclipse.emf:org.eclipse.emf.mwe2.runtime:+'
}
```

**Comment by cdietrich** [comment]
Can you please provide a complete sample project

**Comment by cdietrich** [comment]
Btw using the + might lead to different issues cause wired combinations of Xtext, emf, xcore, platform and jdt will be picked

**Comment by AresEkb** [comment]
Here is a sample project: https://github.com/AresEkb/xcore-gradle-test I think there are a lot of problems with my configuration. For example, GenModelSupport can't be resolved, so I commented it out. I'm trying to resolve errors step by step. I've fixed a problem with an antlr and emf versions. Maybe someone could point me out to a working xcore+gradle sample? This tutorial https://blogs.itemis.com/en/using-xtext-with-xcore-and-gradle seems to be out-dated.
